src/pages/guides/tutorials/stats-addon.md

```diff
@@ -0,0 +1,1093 @@
+import "@spectrum-web-components/table/elements.js";
+import "@spectrum-web-components/button/sp-button.js";
+
+import addOnUISdk from "https://new.express.adobe.com/static/add-on-sdk/sdk.js";
+
+// Wait until the addOnUISdk has been loaded
+addOnUISdk.ready.then(async () => {
+  // API to expose to the Document Sandbox
+  const iframeApi = { 
+    createTable(documentData) { /* ... */ },
+    toggleStatus(sdk) { /* ... */ }
+  };
+  runtime.exposeApi(iframeApi);
+  
+  // Toggle the iframe SDK status light
+  iframeApi.toggleStatus("iframe");
+  
+  // Import the Document Sandbox API proxy
+  const { runtime } = addOnUISdk.instance;
+  const sandboxProxy = await runtime.apiProxy("documentSandbox");
+  // Exposing the iFrame API to the Document Sandbox.
+  
+  // Set the button's click handler
+  const statsButton = document.getElementById("stats");
+  statsButton.addEventListener("click", async () => {
+    // Invoke the Document API via Document Sandbox proxy
+    // to get the document metadata and initiate the table creation process
+    await sandboxProxy.getDocumentData(); // 👈 mind the await
+  });
+
+  // Enable the button only when the addOnUISdk is ready
+  statsButton.disabled = false;
+});
+```
+
+Besides the usual SWC imports, everything must be wrapped by a callback invoked when the `addOnUISdk` module is `ready`—which also means we can switch the first SDK status light to green. Please note that `toggleStatus()` must be available to both the iframe and the Document Sandbox: declaring the `iframeApi` constant first (lines 18-21) and passing it to `exposeApi()` later allows me to call it (line 21).
+
+```js
+// ❌ _in this case_ it's the wrong syntax choice!
+runtime.exposeApi({ 
+  createTable(documentData) { /* ... */ },
+  toggleStatus(sdk) { /* ... */ }
+});
+
+// ✅ better for our needs _in this context_
+const iframeApi = { 
+  createTable(documentData) { /* ... */ },
+  toggleStatus(sdk) { /* ... */ }
+});
+runtime.exposeApi(iframeApi);
+// toggleStatus is accessible in the iframe too
+iframeApi.toggleStatus("iframe"); // 👈
+```
+
+<InlineAlert variant="info" slots="text" />
+
+Let me remind you again of the need to `await` when invoking `getDocumentData()` (line 37, `index.js`), as the Communication API wraps proxy calls with Promises.
+
+In `documentSandbox/code.js`, we bring the iframe proxy in, toggle the status light, and expose the `getDocumentData()` function. Please note that it must be declared asynchronous (line 7) because of the need to `await` when invoking the `panelUIProxy` method `createTable()` (line 12).
+
+<CodeBlock slots="heading, code" repeat="1" languages="code.js"/>
+
+#### documentSandbox/code.js
+
+```js
+import addOnSandboxSdk from "add-on-sdk-document-sandbox";
+const { runtime } = addOnSandboxSdk.instance;
+
+async function start() {
+  const panelUIProxy = await runtime.apiProxy("panel");
+  runtime.exposeApi({
+    async getDocumentData() {                    // 👈 async keyword, because 👇
```

Comment:
One another approach i personally usually follow is keeping the function implementation separate and independent and just use its reference in the `exposeApi`- like 

```ts
function async getDocumentData() {  
... code
}

runtime.exposeApi({
      getDocumentData
});

```

code is more readable that way ?

Reply:
Added that in the info box, too, thanks.